lib/diaspora/mentions_container.rb

```diff
@@ -0,0 +1,37 @@
+module Diaspora
+  module MentionsContainer
+    def self.included(model)
+      model.instance_eval do
+        after_create :create_mentions
+        has_many :mentions, as: :mentions_container, dependent: :destroy
+      end
+    end
```

Comment:
Well, we're inside Rails, so it might be more idiomatic to use `AS::Concern` here.

```rb
  extend ActiveSupport::Concern
  included do
    ....
  end
````